File: tests/go/test_project/helloworld.go
package main

import (
	"fmt"
	"os"
)

func main() {
	message := "Hello, World!"
	fmt.Println(message) // Breakpoint line for debugging
	
	if len(os.Args) > 1 {
		fmt.Printf("Arguments: %v\n", os.Args[1:])
	}
	
	result := add(5, 3)
	fmt.Printf("5 + 3 = %d\n", result)
}

// add returns the sum of two integers
func add(a, b int) int {
	return a + b // Breakpoint line for function debugging
}

File: tests/typescript/test_project/helloworld.ts
/**
 * Greet someone with an optional name.
 * @param name - Optional name to greet
 * @returns Greeting message
 */
function greet(name?: string): string {
 if (name) {
    return `Hello, ${name}!`;
  }
  return 'Hello, World!';
}

// Export for testing
export { greet };

console.log(greet());
console.log(greet("World"));
